edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/TwinsController.cs

```diff
@@ -82,18 +97,47 @@ internal static JRaw GetRawJson(byte[] bytes)
 
         async Task<MethodResult> InvokeMethodAsync(DirectMethodRequest directMethodRequest)
         {
-            Events.ReceivedMethodCall(directMethodRequest, this.identity);
+            Events.ReceivedMethodCall(directMethodRequest);
             IEdgeHub edgeHub = await this.edgeHubGetter;
 
-            using (Metrics.TimeDirectMethod(this.identity.Id, directMethodRequest.Id))
-            {
-                DirectMethodResponse directMethodResponse = await edgeHub.InvokeMethodAsync(this.identity.Id, directMethodRequest);
-                Events.ReceivedMethodCallResponse(directMethodRequest, this.identity);
+            MethodResult methodResult;
+            string currentEdgeDeviceId = edgeHub.GetEdgeDeviceId();
 
-                MethodResult methodResult = GetMethodResult(directMethodResponse);
-
-                return methodResult;
+            if (this.TryGetActorId(out string actorDeviceId, out string actorModuleId))
+            {
+                string actorId = $"{actorDeviceId}/{actorModuleId}";
+
+                if (actorDeviceId == currentEdgeDeviceId)
+                {
+                    IHttpRequestAuthenticator authenticator = await this.authenticatorGetter;
```

Comment:
We should move authentication to a Filter or to the OnActionExecuting method

Reply:
Talked offline, turns out the auth filter approach doesn't work well, and the action filter approach is pretty awkward for passing in all the necessary params.  Leaving this as is for now.